src/components/ItemSummary.tsx

```diff
@@ -0,0 +1,101 @@
+import * as React from 'react';
+import { ItemData } from "../LibraryUtilities";
+import { LibraryContainer } from "./LibraryContainer";
+
+interface ItemSummaryProps {
+    libraryContainer: LibraryContainer,
+    data: ItemData;
+    showDescription: boolean;
+}
+
+interface ItemSummaryStates {
+    hasSummaryData: boolean;
+}
+
+export class ItemSummary extends React.Component<ItemSummaryProps, ItemSummaryStates> {
+    summaryData: any;
+
+    constructor(props: ItemSummaryProps) {
+        super(props);
+        this.state = ({ hasSummaryData: false });
+        this.summaryData = null;
+        this.onDataReceived = this.onDataReceived.bind(this);
+    }
+
+    render() {
+        this.onLibraryItemSummaryExpand();
+
+        let descriptionText = this.props.data.description;
+        let input: JSX.Element[] = [];
+        let output: JSX.Element = null;
+        let description: JSX.Element = null;
+        let icon: JSX.Element = null;
+
+        if (this.state.hasSummaryData) {
+            let inputParameters = this.summaryData.InputParameters;
+            let outputParameters = this.summaryData.OutputParameters;
+            let descriptionTextReceived = this.summaryData.Description;
+
+            for (let inputParameter of inputParameters) {
+                let inputParameterName = inputParameter.Item1
+                if (inputParameterName.length > 0) {
+                    inputParameterName += ": ";
+                }
+
+                input.push(<div className={"IOParameter"}>{inputParameterName}{inputParameter.Item2}</div>);
+            }
+
+            output = <div className={"IOParameter"}>{outputParameters}</div>;
+
+            if (descriptionTextReceived.length > 0) {
+                descriptionText = descriptionTextReceived;
+            }
+        }
+
+        if (!descriptionText) {
+            descriptionText = "No description available";
+        }
```

Comment:
I believe for times when `this.state.hasSummaryData == false`, we should show *Fetching summary...*? This will be cleared anyway when the summary finally arrives.